business_rules/utils.py

```diff
@@ -85,3 +86,64 @@ def params_dict_to_list(params):
             'field_type': param_field_type
         } for name, param_field_type in params.items()
     ]
+
+
+def validate_rule_data(variables, actions, rule):
+    """
+    validate_rule_data is used to check a generated rule against a set of variables and actions
+    :param variables:
+    :param actions:
+    :param rule:
+    :return: bool
+    """
+    def validate_root_keys(rule):
+        root_keys = rule.keys()
+        if 'actions' not in root_keys:
+            raise AssertionError('Missing "{}" key'.format('actions'))
+        if 'conditions' not in root_keys:
+            raise AssertionError('Missing "{}" key'.format('conditions'))
+
+    def validate_condition_operator(condition, rule_schema):
+        for item in rule_schema.get('variables'):
+            if item.get('name') == condition.get('name'):
+                condition_field_type = item.get('field_type')
+                variable_operators = rule_schema.get('variable_type_operators', {}).get(condition_field_type, [])
+                for operators in variable_operators:
+                    if operators['name'] == condition['operator']:
+                        return True
+                raise AssertionError('Unknown operator "{}"'.format(condition['operator']))
+
+    def validate_conditions(input_conditions, rule_schema):
+        if isinstance(input_conditions, list):
+            for condition in input_conditions:
+                validate_conditions(condition, rule_schema)
+        if isinstance(input_conditions, dict):
+            keys = input_conditions.keys()
+            if 'any' in keys or 'all' in keys:
+                if len(keys) > 1:
```

Comment:
If there is `any` or `all` then it only makes sense in business rules to have one key